Makefile

```diff
@@ -184,12 +183,12 @@ test: vendor $(ADDON_ASSET_FILE)
 .PHONY: integration
 integration: $(MINISHIFT_BINARY)
 	mkdir -p $(INTEGRATION_TEST_DIR)
-	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) $(GODOG_OPTS)
+	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) -tags basic $(GODOG_OPTS)
```

Comment:
This won't work for make integration GODOG_OPTS="-tags <feature tag>. 

```
$ make integration GODOG_OPTS="-tags cmd-version"
go install -pkgdir=/home/amit/go/src/github.com/minishift/minishift/out/bindata -ldflags="-X github.com/minishift/minishift/pkg/version.minishiftVersion=1.6.0 -X github.com/minishift/minishift/pkg/version.b2dIsoVersion=v1.2.0 -X github.com/minishift/minishift/pkg/version.centOsIsoVersion=v1.2.0 -X github.com/minishift/minishift/pkg/version.openshiftVersion=v3.6.0 -X github.com/minishift/minishift/pkg/version.commitSha=e6e5ec18" ./cmd/minishift
mkdir -p /home/amit/go/src/github.com/minishift/minishift/out/integration-test
go test -timeout 3600s github.com/minishift/minishift/test/integration --tags=integration -v -args --test-dir /home/amit/go/src/github.com/minishift/minishift/out/integration-test --binary /home/amit/go/bin/minishift -tags basic -tags cmd-version
Test run using Boot2Docker iso image.
Keeping Minishift cache directory '/home/amit/go/src/github.com/minishift/minishift/out/integration-test/cache' for test run.
Log successfully started, logging into: /home/amit/go/src/github.com/minishift/minishift/out/integration-test/integration.log
Running Integration test in: /home/amit/go/src/github.com/minishift/minishift/out/integration-test
Using binary: /home/amit/go/bin/minishift
Feature: Basic
  As a user I can perform basic operations of Minishift and OpenShift

Feature: Addons command and its subcommands

Feature: Minishift config subcommands
  Commands `minishift config [sub-command]` are used for storing
  user defined options which changes default behaviour of Minishift.

Feature: Openshift commands
  Commands "minishift openshift [sub-command]" are used for interaction with Openshift
  cluster in VM provided by Minishift.

Feature: Version command
  Command "minishift version" shows version of minishift binary.

  Scenario: User can get information about version of Minishift                  # features/cmd-version.feature:5
    When executing "minishift version" succeeds                                  # integration_test.go:682 -> executingMinishiftCommandSucceedsOrFails
    Then stdout should match "^minishift v[0-9]\.[0-9]\.[0-9]\+[a-z0-9]{7,8}\n$" # integration_test.go:660 -> commandReturnShouldMatchRegex

Feature: Cool Store
  In order to test Minishift under load
  I need to setup a test environment
  Using CentOS and 4G of memory
  After which I will import image streams
  Finally I will deploy the Cool Store

Feature: Flags
  As a user I can perform advanced operations
  such as starting Minishift with optional features

Feature: Profile
  As a user I can perform basic operations of Minishift with profile feature

Feature: Provision an older major release
  As a user I can provision an older major version of minishift

Feature: Minishift can run behind proxy
  As a user I can use Minishift behind a proxy.
  INFO: Code behind this feature will start a proxy server in background
  on port 8181 and will try to get IP of host machine automatically
  to set MINISHIFT_HTTP_PROXY for this feature only.
  However one can override port number or set IP manually if needed.
  To use custom port number run this test with environmental variable
  INTEGRATION_PROXY_CUSTOM_PORT.
  To use custom IP address please set INTEGRATION_PROXY_CUSTOM_IP
  environmental variable.

1 scenarios (1 passed)
2 steps (2 passed)
536.419061ms
testing: warning: no tests to run
PASS
ok      github.com/minishift/minishift/test/integration 0.582s
✔ ~/go/src/github.com/minishift/minishift [master|✚ 1]
14:59 $

Reply:
@agajdosi Right. However i am confuses over the final argument being parsed.  

```
go test -timeout 3600s ... --binary /home/amit/go/bin/minishift -tags basic -tags cmd-version
```

Reply:
The command line argument should overwrite the default values. IMO it should be like ```go test -timeout 3600s ... --binary /home/amit/go/bin/minishift -tags cmd-version```. Not sure how go dog is handling this. Might be a bug in godog or blessing for us atm ;)

Reply:
@agajdosi @budhrg Can we do it this way. So that we can avoid the extra unwanted tags 
```
GODOG_OPTS ?= -tags basic

.PHONY: integration
integration: $(MINISHIFT_BINARY)
	mkdir -p $(INTEGRATION_TEST_DIR)
ifneq ($(GODOG_OPTS),"")
	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) $(GODOG_OPTS)
endif
```

Reply:
> ifneq ($(GODOG_OPTS),"")
	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) $(GODOG_OPTS)
endif

This will run with default as well when passed via cmd line right as you are checking not equal to.

Reply:
> This will run with default as well when passed via cmd line right as you are checking not equal to.

Yes

Reply:
Correct, `basic` will be overwritten only if user defines her own set of tags.